### DISCUSSION TITLE
Issue inserting a record with 1:many relation with generated resolvers

### DISCUSSION THREAD

**Discussion by gammagandalph** [MichalLytek/typegraphql-prisma#312] [Q&A]
Hi, I am having a bit of trouble with the generated resolvers and I am not sure wether this is me making a mistake somewhere, an issue or just a limitation of the auto generation. 

Given this model in my `schema.prisma`: 
```js
model Absence {
  id             String          @id @default(cuid())
  createdAt      DateTime        @default(now())
  updatedAt      DateTime        @updatedAt
  user           User            @relation(fields: [userId], references: [id])
  userId         String
  absenceType    AbsenceType     @relation(fields: [absenceTypeId], references: [id])
  absenceTypeId  String
  start          DateTime        @db.Date
  end            DateTime        @db.Date
  closed         Boolean?
  updates        AbsenceUpdate[]
  additionalInfo String?         @db.Text
}
```
(omitting the rest of the schema.prisma, but attached in case it's relevant)
I am trying to execute this mutation:
```gql
mutation CreateOneAbsence(
  $start: DateTime!
  $end: DateTime!
  $userId: String
  $absenceTypeId: String
) {
  createOneAbsence(
    data: {
      start: $start
      end: $end
      user: { connect: { id: $userId } }
      absenceType: { connect: { id: $absenceTypeId } }
    }
  ) {
    id
  }
}
```
but I am getting this error: 

Invalid `prisma.absence.create()` invocation:\n\n\n
  Failed to validate the query: `Unable to match input value to any allowed input type for the field. Parse errors: [Query parsing/validation error at `Mutation.createOneAbsence.data.AbsenceCreateInput.start`: A value is required but not set., Query parsing/validation error at `Mutation.createOneAbsence.data.AbsenceUncheckedCreateInput.userId`: A value is required but not set.]` at `Mutation.createOneAbsence.data`

Apologies if this is some dumb beginner's mistake...

**Comment by gammagandalph** [comment]
[schema.prisma.txt](https://github.com/MichalLytek/typegraphql-prisma/files/9166431/schema.prisma.txt)

**Comment by MichalLytek** [comment]
TypeGraphQL-Prisma enforces nested input - `start: { set: "..." }`
See #290

**Reply by gammagandalph** [reply]
I am not sure what to do with this? The mutation I gave I put into the Apollo-Server-Playground. It doesn't seem to have any issue with the start and end values, just with the relations to user and absenceType.

For example, this mutation run on the playground works perfectly fine with the generated resolvers:
```gql
mutation Test($name: String!) {
  createOneAssignment(data: { name: $name }) {
    id
  }
}
```

**Reply by gammagandalph** [reply]
OK, after a bit of playing around, it works if start and end in the schema.prisma are optional.
